### ISSUE TITLE
Feature Request: How to configure DAPR Service Invocation Access Control

### ISSUE THREAD

**Issue by joshuadmatthews** [microsoft/azure-container-apps#303]
**Is your feature request related to a problem? Please describe**.  
A clear and concise description of what the problem is

I need to define access controls for service invocations between dapr services

https://docs.dapr.io/operations/configuration/invoke-allowlist/

**Describe the solution you'd like**.    
A clear and concise description of what you want to happen, any scenarios it will unblock, etc.  

Some way to setup the access controls. This is crucial for preventing a wild wild west scenario of service calls across services.

**Describe alternatives you've considered**.   

Using AKS instead of ACA for full dapr support

**Additional context**.

**Comment by kendallroden** [comment]
This feature hasn't been exposed in container apps yet but we are working towards providing feature parity. We have this in our backlog for evaluation and will report back in the future when there are updates.

**Comment by kendallroden** [comment]
I will try to find a way for us to more transparently share our backlog and work items planned @joshuadmatthews

**Comment by alhardy** [comment]
Hi @kendallroden, can we have an update on this please

**Comment by KamilLach** [comment]
Hello, I know this is old thread, but is there any update on this ? Checked ACA roadmap and didn't find it

**Comment by BlackRider97** [comment]
Thank you for the interest in this feature — this is one of our most-requested Dapr items.

While the exact Dapr [service invocation access control](https://docs.dapr.io/operations/configuration/invoke-allowlist/) (allow/deny lists) configuration is not yet directly exposed in ACA, there are several alternatives available today:

1. **[Dapr component scopes](https://learn.microsoft.com/en-us/azure/container-apps/dapr-component-connection#component-scopes)** — control which apps can access specific components
2. **[VNet integration](https://learn.microsoft.com/en-us/azure/container-apps/vnet-custom)** with Network Security Groups for network-level isolation
3. **App-level middleware** for authorization logic

We're tracking the request for native Dapr Configuration resource support (which would enable invoke allow/deny lists) in our backlog. If this is still a blocking requirement for your scenario, please open a new issue describing your current setup so we can better prioritize.

Closing this long-standing issue — your feedback has been noted.

---
*🤖 This comment was generated with AI assistance and may contain inaccuracies. If this doesn't address your concern or if the issue persists, please reopen this issue or create a new one.*
